### ISSUE TITLE
EnableAnalyticalStorage capability in azurerm_cosmosdb_account doesn't work as expected

### ISSUE THREAD

**Issue by jamesggraf** [terraform-providers/terraform-provider-azurerm#9094]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform: 0.13.3
AzureRM Provider: 2.34.0

### Affected Resource(s)

* `azurerm_cosmosdb_account`

### Terraform Configuration Files

```hcl
resource "azurerm_cosmosdb_account" "db" {
  name                = local.cosmosdb_name
  location            = var.location
  resource_group_name = var.resource_group_name
  offer_type          = "Standard"
  kind                = var.cosmosdb_kind

  capabilities {
    name = "EnableAnalyticalStorage"
  }
}
```

### Expected Behavior

`enableAnalyticalStorage` on CosmosDB account should be set to true

### Actual Behavior

Terraform attempts to destroy and recreate the CosmosDB account even if EnableAnalyticalStorage was already manually enabled on the account through the Azure portal.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

I suspect that this is caused by a change in the Azure API. 10 days ago I opened this ticket requesting this functionality: https://github.com/terraform-providers/terraform-provider-azurerm/issues/8943

When I opened that ticket, Azure/Terraform was complaining that the `EnableAnalyticalStorage` capability was not set in my Terraform on an account that had this capability manually enabled via the Azure Portal by another developer.

Now it appears this feature has migrated from the Cosmos account's list of capabilities, to a flag named `enableAnalyticalStorage`. You can see this by examining the JSON for any CosmosDB account.

### References

* https://docs.microsoft.com/en-us/azure/cosmos-db/cosmosdb-jupyter-notebooks

**Comment by jprendingue** [comment]
I am experiencing the same issue as well. Adding a lifecycle-ignore_changes for capabilities didn't help either.

**Comment by jamesggraf** [comment]
This appears to be addressed with this merge: https://github.com/terraform-providers/terraform-provider-azurerm/pull/10055

**Comment by the-rennegade** [comment]
Our team ran a TF apply at 1:41 PM ET today (01/14/2020) with new provider version (2.43.0) in merge 10055. We did have a capability specified on a Cosmos DB account and also had lifecycle ignore set on the same property. 

The terraform plan did not indicate the resource would be re-created.

Unfortunately, we did not have the expected behavior; nothing being changed on this Cosmos account resource. Instead, the Cosmos resource ended up being destroyed and recreated.

See below for the resource configuration (some lines removed or obfuscated):

`resource "azurerm_cosmosdb_account" "cosmos" {
  for_each = {
    for key, value in var.cosmos :
    key => value
  }

  name                              = each.value.name
  location                          = each.value.location
  resource_group_name               = each.value.resource_group
  offer_type                        = "Standard"
  kind                              = "GlobalDocumentDB"
  is_virtual_network_filter_enabled = true
  ip_range_filter                   = join(",", concat(var.ip_range_azure_portal, each.value.ip_range_filter))

  consistency_policy {
    consistency_level = "Session"
  }

  dynamic "capabilities" {
    for_each = each.value.capabilities
    content {
      name = capabilities.value
    }
  }

  geo_location {
    location          = each.value.location
    failover_priority = 0
  }

  lifecycle {
    ignore_changes = [
      capabilities
    ]
  }
}`

`  CI = {
    name             = "---------"
    resource_group   = "--------"
    location         = "--------"
    vnet_map_key     = "--------"
    akv_key          = "--------"
    la_workspace_key = "--------"
    capabilities     = ["EnableAnalyticalStorage"]
    ip_range_filter  = []
    }`
    
 Would this have been expected behavior?

Thanks!
